manticore/core/cpu/abstractcpu.py

```diff
@@ -324,6 +385,20 @@ def __init__(self, regfile, memory):
         self._md.syntax = 0
         self.instruction = None
 
+        #####################################################
+        # Signals
+        # signal handlers must have this signature:
+        # handler(cpu, *args, **kwargs)
+        self.will_decode_instruction = Signal()
+        self.will_execute_instruction = Signal()
+        self.did_execute_instruction = Signal()
+        self.will_emulate_instruction = Signal()
+        self.did_emulate_instruction = Signal()
+        self.will_read_register = Signal()
```

Comment:
All signals here are at least emitted. 
A use case for the read/write-register/memory could be: report the next time you "see" a tainted value. When a tainted value is read from memory or register.

When we emulate an instruction we lost read/write-register//memory so I assume that a user would want to know when those are not available. How? hooking on will_emulate/did_emulate